.NET/Microsoft.Recognizers.Text.Choice/Models/ChoiceModel.cs

```diff
@@ -20,27 +20,30 @@ protected ChoiceModel(IParser parser, IExtractor extractor)
 
         public List<ModelResult> Parse(string query)
         {
+            var parseResults = Enumerable.Empty<ParseResult>();
+
             try
             {
                 var extractResults = Extractor.Extract(query);
-                var parseResults = extractResults.Select(r => Parser.Parse(r));
-
-                // o can be null, add this part to try.
-                return parseResults.Select(pr => new ModelResult()
-                {
-                    Start = pr.Start.Value,
-                    End = pr.Start.Value + pr.Length.Value - 1,
-                    Resolution = GetResolution(pr),
-                    Text = pr.Text,
-                    TypeName = ModelTypeName,
-                }).ToList();
+                parseResults = extractResults.Select(r => Parser.Parse(r));
+
             }
             catch (Exception)
             {
-                return null;
+                // Nothing to do. Exceptions in parse should not break users of recognizers.
+                // No result.
             }
+
+            return parseResults.Select(pr => new ModelResult()
+            {
+                Start = pr.Start.Value,
+                End = pr.Start.Value + pr.Length.Value - 1,
```

Comment:
Won't this return an invalid object and the consuming code will need to filter these out?